crates/uv-requirements/src/lookahead.rs

```diff
@@ -144,14 +157,30 @@ impl<'a, Context: BuildContext + Send + Sync> LookaheadResolver<'a, Context> {
         &self,
         requirement: Requirement,
     ) -> Result<Option<RequestedRequirements>, LookaheadError> {
-        // Determine whether the requirement represents a local distribution.
-        let Some(VersionOrUrl::Url(url)) = requirement.version_or_url.as_ref() else {
-            return Ok(None);
+        // Determine whether the requirement represents a local distribution and convert to a
+        // buildable distribution.
+        let dist = match requirement.source {
+            RequirementSource::Registry { .. } => return Ok(None),
+            RequirementSource::Url { url, subdirectory } => {
+                let mut merged_url: Url = url.deref().clone();
+                if let Some(subdirectory) = subdirectory {
+                    merged_url
+                        .set_fragment(Some(&format!("subdirectory={}", subdirectory.display())));
+                }
+                let mut merged_url = VerbatimUrl::from_url(merged_url);
+                if let Some(given) = url.given() {
+                    merged_url = merged_url.with_given(given);
+                }
```

Comment:
How can we avoid having to do this conversion here? This seems like the wrong place for this logic. Why should the lookahead resolver need to know how to translate from a URL and subdirectory to a `Dist`?

Reply:
I've front-loaded this by putting a `VerbatimUrl` with the joined url on the url variants of `RequirementSource`.